controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala

```diff
@@ -48,20 +53,33 @@ class ProjectFileController(wdlService: ProjectFileService)(implicit val executi
   }
 
   private def uploadFile(projectId: ProjectId)(implicit accessToken: AccessTokenContent): Route = post {
-    entity(as[ProjectUpdateFileRequest]) { request =>
-      onComplete {
-        for {
-          validateResponse <- wdlService.validateFile(request.projectFile.content)
-          uploadResponse <- wdlService.uploadFile(projectId, request.projectFile, request.version, accessToken.userId)
-        } yield (validateResponse, uploadResponse) match {
-          case (Right(_), Right(_)) => Right(StatusCodes.OK)
-          case (Left(_), Right(_))  => Right(StatusCodes.Created)
-          case (_, Left(response))  => Left(response.getMessage)
-        }
-      } {
-        case Success(Right(sc))   => complete(sc)
-        case Success(Left(error)) => complete(StatusCodes.UnprocessableEntity, s"File have not uploaded due to $error")
-        case Failure(e)           => complete(StatusCodes.InternalServerError, e.getMessage)
+    formFields(Symbol("version"), Symbol("path")) { (version, path) =>
+      fileUpload("file") {
+        case (_, byteSource) =>
+          onSuccess(byteSource.map(_.utf8String).runFold("")((fileContent, string) => fileContent + string)) {
```

Comment:
I'd suggest to change the name of `string: String` to smth more specific. And to use interpolation instead of `fileContent + string`

Reply:
+